fix(builder): print buildpack output, decrease verbosity

gabrtv opened the pull request:
Replaces #59.

gabrtv commented:
Note we still need to cleanup extraneous `mc` output from slugrunner.

arschles commented on rootfs/etc/confd/templates/builder:
```diff
@@ -129,33 +129,35 @@ ACCESS_SECRET=`cat /var/run/secrets/object/store/access-secret-key`
+kubectl --namespace=$POD_NAMESPACE create -f /etc/${SLUG_NAME}.yaml >/dev/null
+
+# wait for pod to be running and then pull its logs
+until [ "`kubectl --namespace=$POD_NAMESPACE get pods -o yaml ${META_NAME} | grep "phase: " | awk {'print $2'}`" == "Running" ]; do
```
print something here to indicate that we've started the build?

sgoings replied:
That `$POD_NAMESPACE` and `$META_NAME` are living dangerously without double quotes around them... but I see lots of other problems in this file. So I'm just going to run away and say this looks good to MEEEEeeeeee...

arschles replied:
@sgoings I'd love you to crush this file with some bash-fu. Feel free to do so (for beta)!

technosophos replied:
I, too, like to live dangerously.
![dangerously](http://cdn.meme.am/instances/500x/57846784.jpg)

sgoings replied:
@gabrtv wants to see it written in !shell which I think is a good idea especially since we could integrate with kubernetes apis here for better? success?

arschles replied:
Yes, this will be rewritten in Go soon: https://github.com/deis/builder/issues/29

arschles commented:
testing locally...

sgoings commented:
Closes #58

arschles commented:
@gabrtv this looks to effectively stream logs from slugbuilder, but it still shows `mc` output (see below). I think the problem is that both stdout and stderr are streamed across the SSH conn at https://github.com/deis/builder/blob/master/pkg/git/git.go#L162-L163. Thoughts on piping `mc` output to `/dev/null` for now?

```console
ENG000656:example-go aaronschlesinger$ echo "asd" >> README.md && git commit -a -m "Asd" && git push deis master
[master 66112e3] Asd
 1 file changed, 1 insertion(+)
Counting objects: 267, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (122/122), done.
Writing objects: 100% (267/267), 3.02 MiB | 0 bytes/s, done.
Total 267 (delta 146), reused 241 (delta 131)
mc: Configuration written to [/var/minio-conf/config.json]. Please update your access credentials.
mc: Successfully created ‘/var/minio-conf/share’.
mc: Initialized share uploads ‘/var/minio-conf/share/uploads.json’ file.
mc: Initialized share downloads ‘/var/minio-conf/share/downloads.json’ file.
Added host ‘http://10.3.0.121:9000’ successfully.
mc: <ERROR> Unable to make bucket ‘http://10.3.0.121:9000/git’. The requested bucket name is not available.
‘nested-underdog.tar.gz’ -> ‘http://10.3.0.121:9000/git/home/nested-underdog:git-66112e38/tar’
Total: 2.97 MB, Transferred: 2.97 MB, Speed: 31.80 MB/s
.8TZRY2JRWMPT6UMXR6I5 gbstrOvotMMcg2sMfGUhA5a6Et/EI5ALtIHsobYk 10.3.0.121:9000
mc: Configuration written to [/app/mc_config/config.json]. Please update your access credentials.
mc: Successfully created ‘/app/mc_config/share’.
mc: Initialized share uploads ‘/app/mc_config/share/uploads.json’ file.
mc: Initialized share downloads ‘/app/mc_config/share/downloads.json’ file.
Added host ‘http://10.3.0.121:9000’ successfully.
‘http://10.3.0.121:9000/git/home/nested-underdog:git-66112e38/tar’ -> ‘/tmp/slug.tgz’
Total: 2.97 MB, Transferred: 2.96 MB, Speed: 62.69 MB/s
8TZRY2JRWMPT6UMXR6I5 gbstrOvotMMcg2sMfGUhA5a6Et/EI5ALtIHsobYk 10.3.0.121:9000
mc: Configuration written to [/app/mc_config/config.json]. Please update your access credentials.
mc: Successfully created ‘/app/mc_config/share’.
mc: Initialized share uploads ‘/app/mc_config/share/uploads.json’ file.
mc: Initialized share downloads ‘/app/mc_config/share/downloads.json’ file.
Added host ‘http://10.3.0.121:9000’ successfully.
‘http://10.3.0.121:9000/git/home/nested-underdog:git-66112e38/tar’ -> ‘/tmp/slug.tgz’
Total: 2.97 MB, Transferred: 2.96 MB, Speed: 62.69 MB/s
-----> Go app detected
```

gabrtv commented:
@arschles hmm i'm not seeing all of that output, but I agree we can shift output to `/dev/null` for now.  I know a lot of that output is from slugbuilder.  One sec..

gabrtv commented:
Shifted some stuff to `/dev/null`.  Here is the new output.  I'm almost positive the `mc` debug you see is coming from slugbuilder:

```
$ git push deis master
Counting objects: 1, done.
Writing objects: 100% (1/1), 178 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
.8TZRY2JRWMPT6UMXR6I5 gbstrOvotMMcg2sMfGUhA5a6Et/EI5ALtIHsobYk 10.199.243.237:9000
mc: Configuration written to [/app/mc_config/config.json]. Please update your access credentials.
mc: Successfully created ‘/app/mc_config/share’.
mc: Initialized share uploads ‘/app/mc_config/share/uploads.json’ file.
mc: Initialized share downloads ‘/app/mc_config/share/downloads.json’ file.
Added host ‘http://10.199.243.237:9000’ successfully.
‘http://10.199.243.237:9000/git/home/node:git-728022b9/tar’ -> ‘/tmp/slug.tgz’
Total: 2.52 KB, Transferred: 0 B, Speed: 0 B/s
-----> Node.js app detected
       
-----> Creating runtime environment
       
       NPM_CONFIG_LOGLEVEL=error
       NPM_CONFIG_PRODUCTION=true
       NODE_ENV=production
       NODE_MODULES_CACHE=true
       
-----> Installing binaries
       engines.node (package.json):  0.12.x
       engines.npm (package.json):   unspecified (use default)
       
       Resolving node version 0.12.x via semver.io...
       Downloading and installing node 0.12.9...
       Using default npm version: 2.14.9
       
-----> Restoring cache
       Skipping cache (new runtime signature)
       
-----> Building dependencies
       Pruning any extraneous modules
       Installing node modules (package.json)
       express@3.1.2 node_modules/express
       ├── methods@0.0.1
       ├── fresh@0.1.0
       ├── range-parser@0.0.4
       ├── cookie-signature@1.0.0
       ├── buffer-crc32@0.2.5
       ├── cookie@0.0.5
       ├── commander@0.6.1
       ├── mkdirp@0.3.5
       ├── debug@2.2.0 (ms@0.7.1)
       ├── send@0.1.0 (mime@1.2.6)
       └── connect@2.7.5 (pause@0.0.1, bytes@0.2.0, buffer-crc32@0.1.1, formidable@1.0.11, qs@0.5.1)
       
-----> Caching build
       Clearing previous node cache
       Saving 1 cacheDirectories (default):
       - node_modules
       
-----> Build succeeded!
       └── express@3.1.2
       
-----> Discovering process types
       Procfile declares types -> web
       Default process types for Node.js -> web
-----> Compiled slug size is 9.8M
8TZRY2JRWMPT6UMXR6I5 gbstrOvotMMcg2sMfGUhA5a6Et/EI5ALtIHsobYk 10.199.243.237:9000
Added host ‘http://10.199.243.237:9000’ successfully.
‘/tmp/slug.tgz’ -> ‘http://10.199.243.237:9000/git/home/node:git-728022b9/push/slug.tgz’
Total: 9.71 MB, Transferred: 9.71 MB, Speed: 32.35 MB/s
-----> Launching... 
       done, node:v19 deployed to Deis

       http://node.localhost

       To learn more, use `deis help` or visit http://deis.io

To ssh://git@deis.gabrtv.io:2222/node.git
   186d974..728022b  master -> master
```

arschles commented:
@gabrtv alrighty. gave them LGTM for this then.
